### PR TITLE
Make regression-test-SPARK-8489 working with scala 2.11

### PR DESCRIPTION
**Pull request by ashangit**

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala [hunk 1]
```diff
@@ -278,8 +283,8 @@ object SparkSQLConfTest extends Logging {
 
         // Always add these two metastore settings at the beginning.
         ("spark.sql.hive.metastore.version" -> "0.12") +:
-        ("spark.sql.hive.metastore.jars" -> "maven") +:
-        filteredSettings
+          ("spark.sql.hive.metastore.jars" -> "maven") +:
```
Comment (superbobry) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala]: Why the extra indentation?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala [hunk 2]
```diff
@@ -37,14 +38,14 @@ import org.apache.spark.sql.types.DecimalType
 import org.apache.spark.util.{ResetSystemProperties, Utils}
 
 /**
- * This suite tests spark-submit with applications using HiveContext.
- */
+  * This suite tests spark-submit with applications using HiveContext.
```
Comment (superbobry) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala]: IIUC Spark preferes JavaDoc conventions over the Scaladoc ones, i.e.

```
/**
 * Foo
 *
```

instead of 

```
/**
  * Foo
  */
```

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala [hunk 3]
```diff
@@ -37,14 +38,14 @@ import org.apache.spark.sql.types.DecimalType
...
+  */
 class HiveSparkSubmitSuite
   extends SparkFunSuite
-  with Matchers
-  with BeforeAndAfterEach
-  with ResetSystemProperties
-  with Timeouts {
+    with Matchers
```
Comment (superbobry) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala]: Unrelated whitespace changes.

**Review by superbobry — COMMENTED** [review]
I can't really comment on anything other than a bunch of minor whitespace quirks, which you're free to ignore.

@AnthonyTruchet could you do a meaningful review? :)

**Review by AnthonyTruchet — COMMENTED** [review]
Can you please explain how the -2.1x suffix gets added, thus requiring this ? Otherwise OK.

**Comment by ashangit** [comment]
Formatting comments taken in account

Currently test.jar has only been compiled for scala 2.10. In order to have tests with scala 2.11 working I had to manually compile sql/hive/src/test/resources/regression-test-SPARK-8489/Main.scala and sql/hive/src/test/resources/regression-test-SPARK-8489/MyCoolClass.scala with scala 2.11 and create the test-2.11.jar with those comiled classes.

Then to ensure that the appropriate jar is loaded depending to the scala release we used the scala version to load the good jar

**Review by AnthonyTruchet — APPROVED** [review]
